api/src/main/java/org/pmiops/workbench/db/jdbc/ReportingQueryService.java

```diff
@@ -27,7 +34,89 @@
 
   List<ReportingUser> getUsers();
 
-  List<ReportingWorkspace> getWorkspaces();
-
   List<ReportingWorkspaceFreeTierUsage> getWorkspaceFreeTierUsage();
+
+  List<ReportingWorkspace> getWorkspaces(long limit, long offset);
+
+  // return all workspaces (replaces previous version)
```

Comment:
nit: would drop this comment , no one will know what "previous version" refers to going forward (I realize you probably didn't write this)